assets/js/googlesitekit/modules/datastore/sharing-settings.test.js

```diff
@@ -0,0 +1,367 @@
+									name: 'Search Console',
+									storeName: 'modules/search-console',
+								},
+								{
+									slug: 'pagespeed-insights',
+									name: 'PageSpeed Insights',
+									storeName: 'modules/pagespeed-insights',
+								},
+							],
+							status: 200,
+						}
+					);
+
+					fetchMock.get(
+						/^\/google-site-kit\/v1\/core\/site\/data\/settings/,
+						{ body: { setting1: 'value' }, status: 200 }
+					);
+
+					fetchMock.get(
+						/^\/google-site-kit\/v1\/modules\/search-console\/data\/settings/,
+						{ body: { setting1: 'value' }, status: 200 }
+					);
+
+					fetchMock.get(
+						/^\/google-site-kit\/v1\/modules\/pagespeed-insights\/data\/settings/,
+						{ body: { setting1: 'value' }, status: 200 }
+					);
+
+					fetchMock.postOnce(
+						/^\/google-site-kit\/v1\/core\/modules\/data\/sharing-settings/,
+						{
+							body: {
+								settings: sharingSettings.settings,
+								newOwnerIDs,
+							},
+						}
+					);
+
+					const initialModules = registry
+						.select( CORE_MODULES )
+						.getModules();
+					// The modules info will be its initial value while the modules
+					// info is fetched.
+					expect( initialModules ).toBeUndefined();
+					await untilResolved( registry, CORE_MODULES ).getModules();
+
+					registry
+						.dispatch( CORE_MODULES )
+						.receiveGetSharingSettings( sharingSettings.settings );
+
+					await registry
+						.dispatch( CORE_MODULES )
+						.saveSharingSettings();
+
+					// Ensure the API call was made.
+					expect( fetchMock ).toHaveFetched(
+						/^\/google-site-kit\/v1\/core\/modules\/data\/sharing-settings/
+					);
+
+					expect( fetchMock ).toHaveFetched(
+						/^\/google-site-kit\/v1\/core\/modules\/data\/list/
+					);
+
+					// Ensure the `setOwnerID` action is dispatched and set the ownerID in state
+					// OR not based on the `newOwnerIDs` availability.
+					expect(
+						registry.select( MODULES_SEARCH_CONSOLE ).getOwnerID()
+					).toBe( ownerID );
+					expect(
+						registry
+							.select( MODULES_PAGESPEED_INSIGHTS )
+							.getOwnerID()
+					).toBe( ownerID );
+				}
+			);
+		} );
+
+		describe( 'receiveGetSharingSettings', () => {
+			it( 'requires the sharingSettings param', () => {
+				expect( () => {
+					registry
+						.dispatch( CORE_MODULES )
+						.receiveGetSharingSettings();
+				} ).toThrow( 'sharingSettings is required' );
+			} );
+
+			it( 'receives sharingSettings and sets it to the state', () => {
+				const state = {
+					...store.getState().sharingSettings,
+					...store.getState().savedSharingSettings,
+				};
+
+				registry
+					.dispatch( CORE_MODULES )
+					.receiveGetSharingSettings( sharingSettings.settings );
+
+				expect( store.getState().sharingSettings ).toMatchObject( {
+					...state,
+					...sharingSettings.settings,
+				} );
+				expect( store.getState().savedSharingSettings ).toMatchObject( {
+					...state,
+					...sharingSettings.settings,
+				} );
```

Comment:
Also unnecessary complicated. Just dispatch the `receiveGetSharingSettings` action and check that the state contains the same `sharingSettings` and `savedSharingSettings` objects.